Newsletter vol.5 (#4836)
line/armeria: site/src/pages/news/20230414-newsletter-5.mdx

---
date: 2023-04-20
---

# Armeria Newsletter vol. 5

<video class="hideOnReducedMotion" src="../../images/armeria.m4v" preload="none" autoplay muted loop style="width: 282px; height: 112px;">
  <img src="../../images/armeria.gif" loading="lazy" width="282" height="112" />
</video>

## Announcements

<div class="noBullet">

We have some announcements to share with you.

> - 📢 **Discontinuance of the virtual office hour**:

Reviewer: ```suggestion
- 📢 **Discontinuance of virtual office hours**:
```

  - We will transform the existing virtual hours to another form such as quarterly webinar. But we haven't decided how often we hold the webinar or what to cover. Please stay tuned.
- 🙌 **[Armeria adopters page](https://github.com/line/armeria/discussions/4626) and limited goods package**:

>   - You can now show off how well your service take adventage of Armeria!

Reviewer: ```suggestion
  - You can now show off how well your service take advantage of Armeria!
```

  - Don't forget to request the limited goods packages.
<img src="../../images/limited_package.jpg" />
- 🚛 **Moving to Discord**:

>   - We dicided to move to Discord to preserve all the past messages.

Reviewer: "past messages" can be counted

```suggestion
  - We dicided to move to Discord to preserve all past messages.
```

  - But no rush! We'll run the both Slack and Discord for a while. Please join [our Discord server](https://discord.gg/7FH8c6npmg) whenever you're ready.
- 🌟 **We're now 4K family members!**
  - It's amazing to see the steady growth of stars. Thank y'all.
<img src="../../images/4k-stars.png" width="214" />

</div>

## From the devs

<div class="noBullet">

- ✨ We've released several new versions since our last newsletter with a bunch of new exciting features! Here are the highlights between [1.17.1](https://armeria.dev/release-notes/1.17.1) and [1.23.1](https://armeria.dev/release-notes/1.23.1).
  - Resilience4j circuit breaker support
  - Improvements in gRPC support
    - Asynchronous ServerInterceptor + Kotlin Coroutine support

>     - Annotations support

Reviewer: ```suggestion
    - Improved annotations support
```

  - Implement Armeria gRPC service and client
  - Exception handling
  - Invoke gRPC service via DocService
- 🔋 Here are some cool open source projects powered by Armeria
  - [Thorium Framework](https://github.com/line/armeria/discussions/4626#discussioncomment-4668166)
    - A Scala 3 microservice framework leveraging on Armeria
  - [Apache Skywalking](https://github.com/line/armeria/discussions/4626#discussioncomment-4863434)
    - An application performance monitor tool for distributed systems built on top of Armeria
  - [Scavenger](https://github.com/naver/scavenger)
    - A runtime dead code analysis tool on top of Armeria

</div>

## We need your comments

<div class="noBullet">

- 🤔 Your opinion means a lot to us. Please let us know what you think about these proposals:
  - #4314 Provide an easier way to get a ClientRequestContext
  - #4569 Consider RFC 7807 Problem Details for HTTP APIs as the default response format in REST APIs
  - #4738 Consider a way to make OAuth2AuthorizaitonGrant dynamic when update the grant information needed

</div>

## Good first issues

<div class="noBullet">

> 🔰 Want to contribute but not sure where to start from? Try one of these:

Reviewer: ```suggestion
🔰 Want to contribute but not sure where to start from? Try this issue:
```
